app/jobs/send_client_ctc_experience_survey_job.rb

```diff
@@ -0,0 +1,16 @@
+class SendClientCtcExperienceSurveyJob < ApplicationJob
+  def perform(client)
+    client.with_lock do
```

Comment:
i feel like it would make more sense to me to have the survey sender deal with assigning a variant instead of doing it in this job since it's a survey-level concern. It's not immediately clear from reading this code where the survey_variant gets used, so having it live closer to where the data is consumed might help.